virttest/bootstrap.py

```diff
@@ -365,10 +370,11 @@ def _forced_or_detected(forced, detected):
         cfg.write("                            - @%s:\n" % release)
         cfg.write("                                variants:\n")
         cfg.write("                                    - @%s:\n" % arch)
-        cfg.write("variants:\n")
-        cfg.write("    - @HostCpuVendor:\n")
-        cfg.write("        variants:\n")
-        cfg.write("            - @%s:\n" % vendor)
+        if vendor:
+            cfg.write("variants:\n")
+            cfg.write("    - @HostCpuVendor:\n")
+            cfg.write("        variants:\n")
+            cfg.write("            - @%s:\n" % vendor)
```

Comment:
Btw similar problem is with the `cpu_version`, but as mentioned in review it would just report `None` and not crash so I don't think it ought to be treated. Anyway feel free to force-push in case you consider it a real issue.

Reply:
I prefer to handle the None `version` too, a null variant is meaningless:
```
- @HostCpuVendor:
    variants:
        - @:
```

Reply:
@ldoktor @yanan-fu no issues for cpu_version as it is already handled with check 
```if cpu_version:
                cfg.write("                variants:\n")
                cfg.write("                    - @HostCpuVersion:\n")
                cfg.write("                        variants:\n")
                cfg.write("                            - @%s:\n" % cpu_version)
```
and we are good

Reply:
Ah~  yes, did not expand all, so, Ack , thanks~